### PR TITLE
release-1.7: Update boilerplate for 2018

### PR DESCRIPTION
**Pull request by nikhita**
This is acts a cherry pick of https://github.com/kubernetes/kubernetes/pull/57735.

Note: This is _not_ an automated cherry-pick since **lots** of things have changed since 1.7 and a cherry-pick leads to **lots** of conflicts in the generated files.

This PR has been created by manually making the changes as in https://github.com/kubernetes/kubernetes/pull/57735/files/83bc631bee60ef2c6f9024869a8747ba9eddf556 and then running `hack/update-all.sh`.

**This also unblocks further client-go 4.0 releases**. (see https://github.com/kubernetes/kubernetes/pull/57245)

**Release note**:

```release-note
NONE
```

/assign wojtek-t cblecker sttts

### CODE REVIEW

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
    "title": "Kubernetes",
-   "version": "v1.7.13"
+   "version": "v1.7.12"
```
Comment (nikhita) [inline api/openapi-spec/swagger.json]: Is this correct?

File: staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"ImportPath": "k8s.io/apiextensions-apiserver",
-	"GoVersion": "go1.8",
+	"GoVersion": "go1.9",
```
Comment (nikhita) [inline staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json]: Is this correct?

File: staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json [hunk 3]
```diff
@@ -174,6 +174,10 @@
 			"ImportPath": "github.com/golang/protobuf/ptypes/timestamp",
 			"Rev": "4bd1920723d7b7c925de087aa32e2187708897f7"
 		},
+		{
+			"ImportPath": "github.com/google/btree",
+			"Rev": "7d79101e329e5a3adf994758c578dab82b90c017"
```
Comment (nikhita) [inline staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json]: Are these changes correct?

File: api/openapi-spec/swagger.json (hunk 1 above)
Comment (wojtek-t) [reply api/openapi-spec/swagger.json]: hmm - I don't think so.

File: federation/apis/openapi-spec/swagger.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
    "title": "Generic API Server",
-   "version": "v1.7.13"
+   "version": "v1.7.12"
```
Comment (wojtek-t) [inline federation/apis/openapi-spec/swagger.json]: this one too

File: staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json]: Did we support 1.9 with kube 1.7? You probably have to switch the go version locally.

**Review by wojtek-t — COMMENTED** [review]
Neither of those non-year related changes look ok.

File: staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json (hunk 3 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json]: This is odd. Is it the only one added? Or anything changed? Maybe godep behaviour is slightly different, or we had invalid Godeps.json? Does our CI test this?

File: staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json (hunk 2 above)
Comment (nikhita) [reply staging/src/k8s.io/apiextensions-apiserver/Godeps/Godeps.json]: > You probably have to switch the go version locally.

Yeah, thought so. Will do.

**Comment by wojtek-t** [comment]
@cblecker - this PR still doesn't seem to be correct. If you can take it over and help with that it would be great.

**Comment by nikhita** [comment]
Closing in favor of https://github.com/kubernetes/kubernetes/pull/57804.

**Pull request closed without merging**